src/3box.js

```diff
@@ -118,6 +130,10 @@ class Box {
           syncPromises.push(this.public._sync(data.numEntries))
           hasResponse[pubStoreAddress] = true
         }
+        if (spaceMessageFilterActive && data.odbAddress.includes('space') === true) {
+          const spaceName = data.odbAddress.split('/')[3].split('.')[2]
+          this.spacesPubSubMessages[spaceName] = data
```

Comment:
It seems like this will just save whatever space with the given name is called last. So if box A & B both open space `S` there could be a problem where A thinks it has the same number of entries as B.
To solve this we likely need to store the entire odbAddress and send spacesPubSubMsgs along to the `open` method.

Reply:
ah yes yes, constantly forgetting the message the space is a global space, nice catch, yeah i think the sol you mentioned is the simplest at the moment